systemtest/rum_test.go

```diff
@@ -110,7 +110,7 @@ func TestRUMAllowServiceNames(t *testing.T) {
 
 	respBody, _ := io.ReadAll(resp.Body)
 	assert.Equal(t, http.StatusForbidden, resp.StatusCode, string(respBody))
-	assert.Equal(t, `{"accepted":0,"errors":[{"message":"unauthorized: anonymous access not permitted for service \"disallowed\""}]}`+"\n", string(respBody))
+	assert.Equal(t, `{"accepted":0,"errors":[{"message":"cannot handle stream: cannot process batch: unauthorized: anonymous access not permitted for service \"disallowed\""}]}`+"\n", string(respBody))
```

Comment:
> cannot handle stream: cannot process batch:

It may not be intended that this additional error message be sent back to the client. but we can revisit it later

Reply:
Those additional error context come from this PR https://github.com/elastic/apm-data/pull/181

The goal was to provide additional information to better troubleshoot errors, but they were not intended to be consumed by end clients.

Reply:
@endorama if the additional information is not intended to be surfaced, what would be your suggestion then on how to handle the enriched error in higher level libraries or applications? IMO it would be quite tedious to parse error responses and parse them to only return a certain error part to the caller.

Reply:
Looking at the code, the function that process errors before responding to clients is `processStreamError`
https://github.com/elastic/apm-server/blob/0f695473c7c2686ac05ca95622aa46675ac7243a/internal/beater/api/intake/handler.go#L172-L210

My understanding is that so far we have been ok in returning error messages from application code as they are, as the error response message is built using `err.Error()` in all cases.
If we consider this not acceptable we should be able to overcome this by either:
- wrap errors where they are handled in `apm-server` code so that `processStreamError` can then return only the meaningful part of it or a specific error message
- unwrap all wrapped errors up to the original error (es pseudocode: `while unwrap(e) != nil; msg = e.Error`); this may have performance downsides, not sure how impactful
- add more error types in `apm-data` so we can handle errors with more granularity

Reply:
While unwrapping errors is possible, IMO it's not great to have the apm-server be responsible for this. The server doesn't know which layer contains potentially sensitive information and which doesn't.

I don't think that the currently returned additional message is a concern, so I don't see an issue with moving forward with this.

Reply:
I agree on the server not knowing which layer contains potential sensitive information, I think this concern has not been raised in this case because the message is more verbose but does not include sensitive information.